sentry-android-core/src/main/java/io/sentry/android/core/performance/AppStartMetrics.java

```diff
@@ -195,10 +222,57 @@ public static void onApplicationCreate(final @NotNull Application application) {
     final @NotNull AppStartMetrics instance = getInstance();
     if (instance.applicationOnCreate.hasNotStarted()) {
       instance.applicationOnCreate.setStartedAt(now);
+      application.registerActivityLifecycleCallbacks(instance);
```

Comment:
this looks way better - thanks! However, I think this is only executed for the case where our Gradle plugin instrumentation is applied, but not for other cases (e.g. no gradle plugin, or SentryPerformanceProvider removed, or enablePerformanceV2 is disabled, etc.). I feel like we should register for callbacks in the AppStartMetrics constructor, or something like that to cover all the cases

Reply:
The problem is that I need an application instance to register the callbacks
I should add another API for that, and maybe that's an overkill
wdyt?

Reply:
adding another method that accepts Application instance is fine I think

Reply:
I added the foreground check in the constructor, though, so that should already improve those cases

Reply:
yeah, but it's still incorrect, foreground importance doesn't always catch background starts. I think with only this change we're not event fixing #3329 properly, because I don't think perfV2 is enabled there, looking at the screenshot. So every background start taking less than 1 minute will still be reported as if it was a proper app start

Reply:
actually, right now we are not even checking the foreground flag for some reason 😅 
this means adding a manual API, unless we call it automatically in our contentProvider.
That would leave out only the case where the user doesn't use the gradle plugin and removes the content provider, but there wouldn't be the need to call a manual API, and imho it should be good enough